src/benchmarklib/tpcc/tpcc_table_generator.cpp

```diff
@@ -513,28 +513,27 @@ std::unordered_map<std::string, BenchmarkTableInfo> TPCCTableGenerator::generate
   Assert(!_benchmark_config->cache_binary_tables, "Caching binary tables is not yet supported for TPC-C");
 
   std::vector<std::thread> threads;
-  auto item_table = std::async(std::launch::async, &TPCCTableGenerator::generate_item_table, this);
-  auto warehouse_table = std::async(std::launch::async, &TPCCTableGenerator::generate_warehouse_table, this);
-  auto stock_table = std::async(std::launch::async, &TPCCTableGenerator::generate_stock_table, this);
-  auto district_table = std::async(std::launch::async, &TPCCTableGenerator::generate_district_table, this);
-  auto customer_table = std::async(std::launch::async, &TPCCTableGenerator::generate_customer_table, this);
-  auto history_table = std::async(std::launch::async, &TPCCTableGenerator::generate_history_table, this);
-  auto order_line_counts = std::async(std::launch::async, &TPCCTableGenerator::generate_order_line_counts, this).get();
-  auto order_table = std::async(std::launch::async, &TPCCTableGenerator::generate_order_table, this, order_line_counts);
-  auto order_line_table =
-      std::async(std::launch::async, &TPCCTableGenerator::generate_order_line_table, this, order_line_counts);
-  auto new_order_table = std::async(std::launch::async, &TPCCTableGenerator::generate_new_order_table, this);
-
-  return std::unordered_map<std::string, BenchmarkTableInfo>(
-      {{"ITEM", BenchmarkTableInfo{item_table.get()}},
-       {"WAREHOUSE", BenchmarkTableInfo{warehouse_table.get()}},
-       {"STOCK", BenchmarkTableInfo{stock_table.get()}},
-       {"DISTRICT", BenchmarkTableInfo{district_table.get()}},
-       {"CUSTOMER", BenchmarkTableInfo{customer_table.get()}},
-       {"HISTORY", BenchmarkTableInfo{history_table.get()}},
-       {"ORDER", BenchmarkTableInfo{order_table.get()}},
-       {"ORDER_LINE", BenchmarkTableInfo{order_line_table.get()}},
-       {"NEW_ORDER", BenchmarkTableInfo{new_order_table.get()}}});
+  auto item_table = generate_item_table();
```

Comment:
Curious: has this been an issue with asynchronous creation?

Reply:
Did Github hide my comment again? :(
TSan didn't like it. I can't find anything wrong with it, but as long as we don't use TPC-C frequently and this only saves a couple of seconds, I rather be on the save side.

Reply:
I was reviewing when you added the comment. Saw it afterwards and makes sense.